tcms/static/js/utils.js

```diff
@@ -1,20 +1,38 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
+	currentGroup = '';
+	// check if using multiple select
+	isMultiple = _select_tag.attributes.getNamedItem('multiple') !== null;
+    
+	// in some cases using single select, the 1st <option> element is ---
     // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
+    if (!isMultiple && _select_tag.options.length) {
         new_options = _select_tag.options[0].outerHTML;
     }
+	
+	data.forEach(function(element) {
+		if(isMultiple && group_attr != null && currentGroup !== element[group_attr]) {
+			if (currentGroup !== '') {
+				// for all but the first time group changes, add a closing optgroup tag
+				new_options += '</optgroup>'
+			}
+			new_options += '<optgroup label="' + element[group_attr] + '">';
+			currentGroup = element[group_attr];
+		}
 
-    data.forEach(function(element) {
         new_options += '<option value="' + element[id_attr] + '">' + element[value_attr] + '</option>';
     });
 
+	// add a final closing optgroup tag if using multipleselect
+	if(isMultiple) {
```

Comment:
Change to `new_options.indexOf('optgroup') > -1` to avoid adding a closing `</optgroup>` tag when no open tags were added before. Can be reproduced by selecting a product which doesn't have any test plans.

Also happens during `loadInitialProduct()`

Reply:
Got it. Made this change